cumulus/xcm/xcm-emulator/src/lib.rs

```diff
@@ -214,12 +213,25 @@ pub trait NetworkComponent {
 	}
 }
 
+pub trait OnHooks {
+	fn on_initialize_hook(n: BlockNumber) -> Weight;
+	fn on_finalize_hook(n: BlockNumber);
```

Comment:
```suggestion
	fn on_initialize(n: BlockNumber) -> Weight;
	fn on_finalize(n: BlockNumber);
```

Reply:
Better

Reply:
I changed the name because there was a namespace collision. A function called `on_initialize` already exists within the scope of the macro, on line 667:
```rust
<<Self as Parachain>::ParachainSystem as Hooks<$crate::BlockNumber>>::on_initialize(block_number);
```
How about updating this function name to `on_initialize_parachain`? Then we can use your suggestion.

Reply:
There should not be any collision if you properly select what trait you are referring to when calling the method. You should be able to rename the methods as suggested.

Reply:
Calling it like this
```rust
<$name as Chain>::Hooks::on_initialize(block_number);
```

Yields the error:
```
error[E0034]: multiple applicable items in scope
   --> cumulus/parachains/integration-tests/emulated/common/src/lib.rs:101:1
    |
101 | / decl_test_parachains! {
102 | |     // Westend Parachains
103 | |     pub struct AssetHubWestend {
104 | |         genesis = asset_hub_westend::genesis(),
...   |
263 | |     }
264 | | }
    | |_^ multiple `on_initialize` found
```

Reply:
Works, thanks!